src/Symfony/Component/Message/Handler/MessageHandlerCollection.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Message\Handler;
+
+/**
+ * Represents a collection of message handlers.
+ *
+ * @author Samuel Roze <samuel.roze@gmail.com>
+ */
+class MessageHandlerCollection
+{
+    /**
+     * @var callable[]
+     */
+    private $handlers;
+
+    /**
+     * @param callable[] $handlers
+     */
+    public function __construct(array $handlers)
+    {
+        if (empty($handlers)) {
+            throw new \InvalidArgumentException('A collection of message handlers requires at least one handler');
+        }
+
+        $this->handlers = $handlers;
+    }
+
+    public function __invoke($message)
+    {
+        return array_map(function ($handler) use ($message) {
+            return $handler($message);
+        }, $this->handlers);
```

Comment:
wouldn't an `array_walk` be better ? Don't see why we would need to fetch the return values of each handlers..

Reply:
As [discussed here](https://github.com/symfony/symfony/pull/24411#discussion_r143650179) I think we should have all we need to get the result of handlers, to support queries. Therefore, `array_map` makes sense here :)